docs/DEVGUIDE.md

```diff
@@ -1,5 +1,24 @@
 # Developer's Guide to Service-Catalog
 
+## Overview
+
+This repository is organized as similarly to Kubernetes itself as the developers
+have found possible (or practical). Below is a summary of the repository's
+layout:
+
+    .
+    ├── .glide         # Glide cache (untracked)
+    ├── bin            # Destination for binaries compiled for linux/amd64 (untracked)
+    ├── build          # Contains build-related scripts and subdirectories containing Dockerfiles
+    ├── cmd            # Contains "main" Go packages for each service catalog component binary
+    ├── contrib        # Contains all non-essential source
+    ├── deploy
+    │   └── catalog    # Helm chart for deploying service catalog
+    ├── docs           # Documentation
+    ├── hack           # Contains scripts useful for development
```

Comment:
Ok... good question. I've worked on the build process a lot these last few days and never had to touch anything in `hack`. I have to admit to a less than perfect understanding of what's in there... so maybe the description I have used on this line is not accurate. Maybe @pmorie can shed light on this. @pmorie... what's the distinction (upstream in k8s) between that which is in `build` and that which is in `hack`?